scipy/stats/tests/test_survival.py

```diff
@@ -1,7 +1,39 @@
 import pytest
 import numpy as np
-from numpy.testing import assert_equal
+from numpy.testing import assert_equal, assert_allclose
 from scipy import stats
+from scipy.stats import _survival
+
+
+def _kaplan_meier_reference(times, censored):
+    # This is a very straightforward implementation of the Kaplan-Meier
+    # estimator that does almost everything differently from the implementation
+    # in stats.ecdf.
+
+    # Begin by sorting the raw data. Note that the order of death and loss
+    # at a given time matters: death happens first. See [2] page 461:
+    # "These conventions may be paraphrased by saying that deaths recorded as
+    # of an age t are treated as if they occurred slightly before t, and losses
+    # recorded as of an age t are treated as occurring slightly after t."
+    # We implement this by sorting the data first by time, then by `censored`,
+    # (which is 0 when there is a death and 1 when there is only a loss).
```

Comment:
To more closely match `ecdf`'s use of `CensoredData`, the second argument is whether the observation is censored (opposite of death), so you need to pass `_kaplan_meier_reference(t2, np.logical_not(d2))`. 

However, the references show tables with "death", so it is easier to compare against the references if these lists are in terms of "death" (instead of censored).